src/middlewared/middlewared/plugins/disk_/wipe.py

```diff
@@ -119,25 +120,18 @@ def _wipe_impl(self, job, dev, mode, event):
                         return
                     job.set_progress(round(((i / iterations) * 100), 2))
 
-    @accepts(
-        Str('dev'),
-        Str('mode', enum=['QUICK', 'FULL', 'FULL_RANDOM'], required=True),
-        Bool('synccache', default=True),
+    @api_method(
+        DiskWipeArgs,
+        DiskWipeResult,
+        roles=["FULL_ADMIN"],
```

Comment:
We have a `DISK_WRITE` role that should probably be used here in lieu of FULL_ADMIIN.

Reply:
It might not be a bad idea to add auditing for this method so that we have history of it being called on a disk.